manifests/agent/service.pp

```diff
@@ -4,10 +4,12 @@
   case $::puppet::runmode {
     'service': {
       service {'puppet':
-        ensure    => running,
-        name      => $puppet::service_name,
-        hasstatus => true,
-        enable    => true,
+        ensure     => running,
+        name       => $puppet::service_name,
+        hasstatus  => true,
+        hasrestart => true,
+        enable     => true,
+        restart    => '/usr/sbin/service puppet reload',
```

Comment:
I don't think this works on windows. Agent has a decent support matrix so maybe this should be in params.pp and undef for most OS-es.